### PR TITLE
tx-generator: derive Plutus budged automatically

### PR DESCRIPTION
**Pull request by MarcFontaine**
This PR adds a plutusAutoCosts mode to the tx-generator.
This mode determines the Plutus script budget by pre-execution of the script,
such that it in no longer necessary to provide values
for the Plutus execution units in definition of benchmark run.

Also:

* Add Plutus loop example.
* Log addresses in splitting phase (useful for debugging).

### CODE REVIEW

**Comment by MarcFontaine** [comment]
Sorry guys, I do not really need a code review from everybody..
(I just clicked a wrong button on Github.)
Just one code-owner review is enough.

**Review by deepfire — APPROVED** [review]

**Pull request merged**
